OBSDOCS-1723: Port the "Installing Logging" sections to 5.8 and 6.y docs (#89053)
openshift/openshift-docs: modules/log6x-logging-installing-logging-operator.adoc

kind: Subscription
metadata:
  name: cluster-logging
  namespace: openshift-logging # <1>
spec:
  channel: stable-6.<y> # <2>
  name: cluster-logging
  source: redhat-operators # <3>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify `stable-6.<y>` as the channel.
<3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object you created when you configured the Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create an `OperatorGroup` object:
+
.Example `OperatorGroup` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: cluster-logging
  namespace: openshift-logging # <1>
spec:
  targetNamespaces:
  - openshift-logging
----
<1> You must specify the `openshift-logging` namespace.

. Apply the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

> 

Reviewer: ```suggestion
. Create a service account to be used by the log collector:
+
[source,terminal]
----
$ oc create sa logging-collector -n openshift-logging
----
. Assign the necessary permissions to the service account for the collector to be able to collect and forward logs:
+
[source,terminal]
----
$ oc adm policy add-cluster-role-to-user logging-collector-logs-writer -z logging-collector -n openshift-logging
$ oc adm policy add-cluster-role-to-user collect-application-logs -z logging-collector -n openshift-logging
$ oc adm policy add-cluster-role-to-user collect-infrastructure-logs -z logging-collector -n openshift-logging
----
```

<5> Point the ClusterLogForwarder to the LokiStack instance.
<6> Select the log output types you want to send to the LokiStack instance.

. Apply the `ClusterLogForwarder CR` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Verify the installation by running the following command:
+
[source,terminal]
----
$ oc get pods -n openshift-logging
----
+
.Example output
[source,terminal]
----
$ oc get pods -n openshift-logging
NAME                                               READY   STATUS    RESTARTS   AGE
cluster-logging-operator-fb7f7cf69-8jsbq           1/1     Running   0          98m
collector-222js                                    2/2     Running   0          18m
collector-g9ddv                                    2/2     Running   0          18m
collector-hfqq8                                    2/2     Running   0          18m
collector-sphwg                                    2/2     Running   0          18m
collector-vv7zn                                    2/2     Running   0          18m
collector-wk5zz                                    2/2     Running   0          18m

> logging-view-plugin-6f76fbb78f-n2n4n               1/1     Running   0          18m

Reviewer: Still not COO installed, this was a typo from my earlier comment
```suggestion
```

// Module is included in the following assemblies:
//
// 
:_mod-docs-content-type: PROCEDURE
[id="log6x-installing-logging-operator_{context}"]
= Installing {clo} by using the CLI

> After you have installed the log store, install {clo}, which collects and forwards the logs to the log store. This can be done from the {product-title} command-line interface (CLI).

Reviewer: Similar comment to the one I did for Loki
```suggestion
After you have installed the log store, install {clo}, which collects and forwards the logs to the log store.
```

.Prerequisites

* You have administrator permissions.
* You installed the {oc-first}.
* You installed and configured {loki-op}.

.Procedure

. Create a `Subscription` object for {clo}:
+
.Example `Subscription` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: cluster-logging
  namespace: openshift-logging # <1>

> spec:
>   channel: stable-6.<y> # <2>
>   name: cluster-logging
>   source: redhat-operators # <3>
>   sourceNamespace: openshift-marketplace
> ----
> <1> You must specify the `openshift-logging` namespace.
> <2> Specify `stable-6.<y>` as the channel.
> <3> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object you created when you configured the Operator Lifecycle Manager (OLM).

Reviewer: ```suggestion
spec:
  channel: stable-6.<y> # <2>
  installPlanApproval: Automatic # <3>
  name: cluster-logging
  source: redhat-operators # <4>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify `stable-6.<y>` as the channel.
<3> If the approval strategy in the subscription is set to `Automatic`, the update process initiates as soon as a new Operator version is available in the selected channel. If the approval strategy is set to `Manual`, you must manually approve pending updates.
<4> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object you created when you configured the Operator Lifecycle Manager (OLM).
```

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create an `OperatorGroup` object:
+
.Example `OperatorGroup` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup

> metadata:
>   name: cluster-logging
>   namespace: openshift-logging # <1>
> spec:
>   targetNamespaces:
>   - openshift-logging

Reviewer: ```suggestion
metadata:
  annotations:
    olm.providedAPIs: ClusterLogForwarder.v1.observability.openshift.io,LogFileMetricExporter.v1alpha1.logging.openshift.io
  name: cluster-logging
  namespace: openshift-logging # <1>
spec:
  upgradeStrategy: Default
```

----
<1> You must specify the `openshift-logging` namespace.

. Apply the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a service account to be used by the log collector:
+
[source,terminal]
----
$ oc create sa logging-collector -n openshift-logging
----

. Assign the necessary permissions to the service account for the collector to be able to collect and forward logs:
+
[source,terminal]
----
$ oc adm policy add-cluster-role-to-user logging-collector-logs-writer -z logging-collector -n openshift-logging
$ oc adm policy add-cluster-role-to-user collect-application-logs -z logging-collector -n openshift-logging
$ oc adm policy add-cluster-role-to-user collect-infrastructure-logs -z logging-collector -n openshift-logging
----

. Create a `ClusterLogForwarder` CR:
+
.Example `ClusterLogForwarder` CR
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:

>   name: instance # <1>
>   namespace: openshift-logging # <2>
> spec:
>   serviceAccount:
>     name: logging-collector # <3>
>   outputs:
>   - name: lokistack-out
>     type: lokiStack # <4>
>     lokiStack:
>       target: # <5>
>         name: logging-loki 
>         namespace: openshift-logging
>       authentication:
>         token:
>           from: serviceAccount
>     tls:
>       ca:
>         key: service-ca.crt
>         configMapName: openshift-service-ca.crt
>   pipelines:
>   - name: all-logs
>     inputRefs: # <6>
>     - application
>     - infrastructure
>     outputRefs:
>     - lokistack-out
> ----
> <1> Use the name `instance`.
> <2> You must specify the `openshift-logging` namespace.
> <3> Use the service account created before.
> <4> Select the `lokiStack` output type to send logs to the LokiStack instance.
> <5> Point the ClusterLogForwarder to the LokiStack instance.
> <6> Select the log output types you want to send to the LokiStack instance.

Reviewer: ```suggestion
  name: instance
  namespace: openshift-logging # <1>
spec:
  serviceAccount:
    name: logging-collector # <2>
  outputs:
  - name: lokistack-out
    type: lokiStack # <3>
    lokiStack:
      target: # <4>
        name: logging-loki 
        namespace: openshift-logging
      authentication:
        token:
          from: serviceAccount
    tls:
      ca:
        key: service-ca.crt
        configMapName: openshift-service-ca.crt
  pipelines:
  - name: all-logs
    inputRefs: # <5>
    - application
    - infrastructure
    outputRefs:
    - lokistack-out
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify the name of the service account created before.
<3> Select the `lokiStack` output type to send logs to the `LokiStack` instance.
<4> Point the `ClusterLogForwarder` to the `LokiStack` instance created before.
<5> Select the log output types you want to send to the `LokiStack` instance.
```
